crates/bindings/src/lib.rs

```diff
@@ -45,13 +42,610 @@ pub use timestamp::Timestamp;
+/// For example:
+/// ```ignore
+///  #[table(name = users)]
+///  pub struct User {
+///      #[autoinc]
+///      #[primary_key]
+///      pub id: u32,
+///      #[unique]
+///      pub username: String,
+///      #[index(btree)]
+///      pub popularity: u32,
+///  }
+///  #[reducer]
+///  fn demo(ctx: &ReducerContext) {
+///      let users: users__TableHandle = ctx.db.users();
+///  }
+///  ```
+///
+/// ### `public` and `private`
+///
+/// Tables are private by default. This means that clients cannot read their contents
+/// or see that they exist.
+///
+/// If you'd like to make your table publically accessible by clients,
+/// put `public` in the macro arguments (e.g.
+/// `#[spacetimedb::table(public)]`). You can also specify `private` if
+/// you'd like to be specific.
+///
+/// This is fully separate from Rust's module visibility
+/// system; `pub struct` or `pub(crate) struct` do not affect the table visibility, only
+/// the visibility of the items in your own source code.
+///
+/// ### `index(...)`
+///
+/// You can specify an index on one or more of the table's columns with the syntax:
+/// `index(name = my_index, btree(columns = [a, b, c]))`
+///
+/// You can also just put `#[index(btree)]` on the field itself if you only need
+/// a single-column attribute; see column attributes below.
+///
+/// Multiple indexes are permitted.
+///
+/// You can use indexes to efficiently [`filter`](crate::BTreeIndex::filter) and
+/// [`delete`](crate::BTreeIndex::delete) rows. This is encapsulated in the struct [`BTreeIndex`].
+///
+/// For a table *table* and an index *index*, use:
+/// ```text
+/// ctx.db.{table}().{index}()
+/// ```
+/// to get a [`BTreeIndex`] for a [`ReducerContext`].
+///
+/// For example:
+/// ```ignore
+///
+/// let by_id_and_username: spacetimedb::BTreeIndex<_, (u32, String), _> =
+///     ctx.db.users().by_id_and_username();
+/// ```
+///
+/// ### `scheduled(reducer_name)`
+///
+/// Used to declare a [scheduled reducer](macro@crate::reducer#scheduled-reducers).
+///
+/// The annotated struct type must have at least the following fields:
+/// - `scheduled_id: u64`
+/// - [`scheduled_at: ScheduleAt`](crate::ScheduleAt)
+///
+/// # Column (field) attributes
+///
+/// ### `#[autoinc]`
+///
+/// Creates an auto-increment constraint.
+///
+/// When a row is inserted with the annotated field set to `0` (zero),
+/// the sequence is incremented, and this value is used instead.
+///
+/// Can only be used on numeric types.
+///
+/// May be combined with indexes or unique constraints.
+///
+/// Note that using `#[autoinc]` on a field does not also imply `#[primary_key]` or `#[unique]`.
+/// If those semantics are desired, those attributes should also be used.
+///
+/// <!-- TODO: What happens if a reducer tries to insert a row that has an already-existing unique
+///            auto-inc column? Like, if the user inserts a row ahead of the auto-inc, then
+///            the auto-inc catches up? -->
```

Comment:
Normal unique constraint violation.

Reply:
On first insert, or when the auto inc catches up?

Reply:
When the auto-inc catches up.

Reply:
Added a note about this in place of the TODO.